src/grids/linear_grid.jl

```diff
@@ -1,217 +1,218 @@
-  _zeropad!(ft_work1, u; fillzero=true)
-  @. ft_work1 *= ft⁻¹_shift
-  unsafe_execute!(ft⁻¹.p, ft_work1, out)
-  
-  return out
+function ift!(grid::LinearGrid, out::AbstractVector, u::AbstractVector)
+    return fftshift(fft(u)) .* grid.op.ft_arr[1:length(u)] * grid.d / 2π
 end
 
 """
 Apply inverse FFT of grid to input
 """
-ifft!(grid::LinearGrid, out::AbstractVector, u::AbstractVector
-  ) = out .= @views _ifft!(grid, grid.op.ft⁻¹_work, u)[1:grid.len]
 
-ifft(grid::LinearGrid, u::AbstractVector
-  ) = ifft!(grid, zeros(eltype(grid.op.ft⁻¹_work), grid.len), u)
+ift(grid::LinearGrid, u::AbstractVector) = ift!(grid, zero(u), copy(u))
 
 """
 Internal function that works on padded arrays
 """
 function _convolution!(grid::LinearGrid, out::AbstractVector,
-    u::AbstractVector,v::AbstractVector)
-  (; padded, ft_work1, ft_work2, ft, ft⁻¹, conv_shift) = grid.op
-  
-  assert_applicable(ft⁻¹.p, ft_work2, out)
-  unsafe_execute!(ft, _zeropad!(padded, u), ft_work1)
-  unsafe_execute!(ft, _zeropad!(padded, v), ft_work2)
-  @. ft_work2 *= ft_work1 * conv_shift
-  unsafe_execute!(ft⁻¹.p, ft_work2, out)
-  
-  return out
+    u::AbstractVector, v::AbstractVector)
+    (; padded, ft_work1, ft_work2, ft_p, bft_p, ft_arr) = grid.op
+
+    assert_applicable(bft_p, ft_work2, out)
+    unsafe_execute!(ft_p, _zeropad!(padded, u), ft_work1)
+    unsafe_execute!(ft_p, _zeropad!(padded, v), ft_work2)
+    @. ft_work2 *= ft_work1 * ft_arr
+
+    unsafe_execute!(bft_p, ft_work2, out)
+
+    out .*= grid.d / length(padded)
+
+    return out
 end
 
 """
 Convolution - takes non-padded arrays as input (and output)
 """
 convolution!(grid::LinearGrid, out::AbstractVector,
-  u::AbstractVector,v::AbstractVector
-  ) = out .= @views _convolution!(grid, grid.op.ft⁻¹_work, u, v)[1:grid.len]
+    u::AbstractVector, v::AbstractVector
+) = out .= @views _convolution!(grid, grid.op.bft_work, u, v)[grid.op.data_range]
 
 convolution(grid::LinearGrid, u::AbstractVector, v::AbstractVector
-  ) = convolution!(grid, zero(u), u, v)
+) = convolution!(grid, zeros(eltype(grid.op.padded), length(u)), u, v)
 
 """
 Internal function that works on padded arrays
 """
 function _crosscorrelation!(grid::LinearGrid, out::AbstractVector,
     u::AbstractVector, v::AbstractVector)
-  (; padded, ft_work1, ft_work2, ft, ft⁻¹, corr_shift) = grid.op
-  
-  assert_applicable(ft⁻¹.p, ft_work2, out)
-  unsafe_execute!(ft, _zeropad!(padded, u), ft_work1)
-  unsafe_execute!(ft, _zeropad!(padded, v), ft_work2)
-  @. ft_work2 *= conj(ft_work1 * corr_shift)
-  unsafe_execute!(ft⁻¹.p, ft_work2, out)
-  
-  return out
+    (; padded, ft_work1, ft_work2, ft_p, bft_p, ft_arr) = grid.op
+
+    assert_applicable(bft_p, ft_work2, out)
+    unsafe_execute!(ft_p, _zeropad!(padded, u), ft_work1)
+    unsafe_execute!(ft_p, _zeropad!(padded, v), ft_work2)
+
+    @. ft_work2 *= conj(ft_work1) * ft_arr
+    unsafe_execute!(bft_p, ft_work2, out)
+
+
+    out .*= grid.d / length(padded)
```

Comment:
Performance wise it should be much better to multiply this into ft_work2 before the FT, I don't know if this is bad for the data though. Should not be, but who knows.